lib/api/2.0/users.js

```diff
@@ -49,45 +48,48 @@ var addUser = controller({success: 201}, function(req, res) {
 });
 
 var modifyUser = controller(function(req) {
-    var userObj = _.pick(req.swagger.params.body.value, ['username', 'password', 'role']);
-    if(!req.isAuthenticated) {
-        throw new Errors.UnauthorizedError('Unauthorized');
-    }
+    var name = req.swagger.params.name.value;
+    var userObj = _.pick(req.swagger.params.body.value, ['password', 'role']);
 
-    if(!req.isAuthenticated()) {
-        throw new Errors.UnauthorizedError('Unauthorized');
-    }
+    return Promise.try(function() {
+        if(!req.isAuthenticated || !req.isAuthenticated()) {
```

Comment:
isAuthenticated is setup as a function in passport when a user is authenticated and logged in or absent otherwise.  I suppose something could tamper with that between the autn and here, but then passport wouldn't work and much more than this check would blow up.  How much less would you want me to trust passport?